kernel/src/main/java/org/kframework/lsp/KLanguageServer.java

```diff
@@ -43,6 +43,8 @@ public CompletableFuture<InitializeResult> initialize(InitializeParams initializ
         /* Check if dynamic registration of completion capability is allowed by the client. If so we don't register the capability.
            Else, we register the completion capability.
          */
+        // TODO: check if this needs to be guarded
```

Comment:
Does it need to be guarded?

Reply:
I'm not sure how the registration of new features works between the client and the server.
Some features may be turned on and off while the plugin is activated.
I got the code from another example. I will have to recheck this part, but for now, I want to concentrate on adding the features, at least until the 8th.

Reply:
OK, if it's a case of prioritising getting features implemented quickly before going back and sorting things out 100% properly then that's fine.